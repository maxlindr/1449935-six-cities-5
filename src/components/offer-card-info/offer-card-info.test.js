import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import OfferCardInfo from './offer-card-info';
import offerMock from '../../../__mocks__/offer-mock';

const renderer = new ShallowRenderer();

test(`OfferCardInfo должен корректно отображаться`, () => {
  renderer.render(
      <OfferCardInfo
        offer={offerMock}
        favorites={true}
      />
  );

  const result = renderer.getRenderOutput();
  expect(result).toMatchSnapshot();
});
