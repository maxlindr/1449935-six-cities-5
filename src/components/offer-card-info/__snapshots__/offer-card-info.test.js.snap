// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OfferCardInfo должен корректно отображаться 1`] = `
<div
  className="favorites__card-info place-card__info"
>
  <div
    className="place-card__price-wrapper"
  >
    <div
      className="place-card__price"
    >
      <b
        className="place-card__price-value"
      >
        €
        493
      </b>
      <span
        className="place-card__price-text"
      >
        / night
      </span>
    </div>
    <Memo(Connect(WithUpdateOfferOnFavoriteToggle))
      active={false}
      offer={
        Object {
          "bedrooms": 3,
          "description": "Discover daily local life in city center, friendly neighborhood, clandestine casino, karaoke, old-style artisans, art gallery and artist studio downstairs.",
          "favorite": false,
          "features": Array [
            "Laptop friendly workspace",
            "Washer",
            "Towels",
            "Fridge",
            "Air conditioning",
            "Baby seat",
            "Breakfast",
          ],
          "host": Object {
            "avatar": "img/avatar-angelina.jpg",
            "id": 25,
            "name": "Angelina",
            "super": true,
          },
          "id": "1",
          "location": Object {
            "city": Object {
              "coordinates": Array [
                50.938361,
                6.959974,
              ],
              "name": "Cologne",
              "zoom": 13,
            },
            "coordinates": Array [
              50.957361,
              6.9509739999999995,
            ],
            "zoom": 16,
          },
          "maxAdults": 4,
          "photos": Array [
            "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/15.jpg",
            "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/16.jpg",
            "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/9.jpg",
            "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/10.jpg",
            "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/4.jpg",
            "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/19.jpg",
            "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/14.jpg",
            "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/3.jpg",
            "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/2.jpg",
            "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/11.jpg",
            "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/1.jpg",
            "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/17.jpg",
            "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/5.jpg",
            "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/13.jpg",
          ],
          "premium": false,
          "price": 493,
          "rating": 3.8,
          "thumbnail": "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/14.jpg",
          "title": "The house among olive ",
          "type": "house",
        }
      }
      type="place-card"
    />
  </div>
  <Memo(RatingStars)
    rating={3.8}
    type="place-card"
  />
  <h2
    className="place-card__name"
  >
    <Link
      to="/offer/1"
    >
      The house among olive 
    </Link>
  </h2>
  <p
    className="place-card__type"
  >
    House
  </p>
</div>
`;
